packages/lib/services/interop/InteropService_Importer_Md.ts

```diff
@@ -59,38 +61,66 @@ export default class InteropService_Importer_Md extends InteropService_Importer_
 		}
 	}
 
+	trimAnchorLink(link: string) {
+		if (link.indexOf('#') <= 0) return link;
+
+		const splitted = link.split('#');
+		splitted.pop();
+		return splitted.join('#');
+	}
+
 	/**
 	 * Parse text for links, attempt to find local file, if found create Joplin resource
 	 * and update link accordingly.
 	 */
-	async importLocalFiles(filePath: string, md: string) {
+	async importLocalFiles(filePath: string, md: string, parentFolderId: string) {
 		let updated = md;
 		const fileLinks = unique(markdownUtils.extractFileUrls(md));
 		await Promise.all(fileLinks.map(async (encodedLink: string) => {
 			const link = decodeURI(encodedLink);
-			const attachmentPath = filename(`${dirname(filePath)}/${link}`, true);
-			const pathWithExtension = `${attachmentPath}.${fileExtension(link)}`;
+			// Handle anchor links appropriately
+			const trimmedLink = this.trimAnchorLink(link);
+			const attachmentPath = filename(`${dirname(filePath)}/${trimmedLink}`, true);
+			const pathWithExtension = `${attachmentPath}.${fileExtension(trimmedLink)}`;
 			const stat = await shim.fsDriver().stat(pathWithExtension);
 			const isDir = stat ? stat.isDirectory() : false;
 			if (stat && !isDir) {
-				const resource = await shim.createResourceFromPath(pathWithExtension);
-				// NOTE: use ](link) in case the link also appears elsewhere, such as in alt text
-				const linkPatternEscaped = pregQuote(`](${link})`);
+				const supportedFileExtension = this.metadata().fileExtensions;
+				const resolvedPath = shim.fsDriver().resolve(pathWithExtension);
+				let id: string = '';
+				// If the link looks like a note, then import it
+				if (supportedFileExtension.indexOf(fileExtension(trimmedLink).toLowerCase()) >= 0) {
+					// If the note hasn't been imported yet, do so now
+					if (!this.importedNotes[resolvedPath]) {
+						await this.importFile(resolvedPath, parentFolderId);
+					}
+
+					id = this.importedNotes[resolvedPath].id;
+				} else {
+					const resource = await shim.createResourceFromPath(pathWithExtension);
+					id = resource.id;
+				}
+
+				// NOTE: use ](encodedLink in case the link also appears elsewhere, such as in alt text
+				const linkPatternEscaped = pregQuote(`](${this.trimAnchorLink(encodedLink)}`);
 				const reg = new RegExp(linkPatternEscaped, 'g');
-				updated = updated.replace(reg, `](:/${resource.id})`);
+				updated = updated.replace(reg, `](:/${id}`);
 			}
 		}));
 		return updated;
 	}
 
 	async importFile(filePath: string, parentFolderId: string) {
+		if (this.importedNotes[filePath]) return this.importedNotes[filePath];
+		this.importedNotes[filePath] = { message: 'Note building in progress' };
```

Comment:
I wonder if that might be confusing to have either a note or a message in this object. Maybe you could define a struct that contains a note and a status (either ready or not)?

Reply:
I'm going to re-write this section , this line was a bit of a hack and it doesn't do its job properly. It's meant to prevent cycles, but the current state of the code will cause a crash of there is a cycle (because there is no id here)